folio-spring-base/src/main/java/org/folio/spring/DefaultFolioExecutionContext.java

```diff
@@ -36,8 +36,10 @@ public DefaultFolioExecutionContext(FolioModuleMetadata folioModuleMetadata,
     Map<String, Collection<String>> allHeaders) {
     this.folioModuleMetadata = folioModuleMetadata;
     this.allHeaders = allHeaders;
-    this.okapiHeaders = new HashMap<>(allHeaders);
-    this.okapiHeaders.entrySet().removeIf(e -> !e.getKey().toLowerCase().startsWith(OKAPI_HEADERS_PREFIX));
+    this.okapiHeaders = allHeaders.entrySet().stream()
```

Comment:
This code doesn't prevent duplicate headers. If you add multiple x-okapi-tenant headers, it ignores the case and only keeps one, which might be random.

A possible solution might be to implement a method that adds a header while checking if it already exists in a case-insensitive manner. If it does, the method would replace the existing header with the new one.